libs/sql-completion/src/main/java/io/crate/sql/completion/SqlCompletion.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.sql.completion;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Vocabulary;
+
+import io.crate.sql.parser.CaseInsensitiveStream;
+import io.crate.sql.parser.antlr.v4.SqlBaseLexer;
+import io.crate.sql.parser.antlr.v4.SqlBaseParser;
+
+public class SqlCompletion {
+
+    public SqlCompletion() {
+    }
+
+    public Iterable<String> getCandidates(String statement) {
```

Comment:
This should later be extended to also take line-nr and column-nr, so that the cursor position can be somewhere within the statement instead of always at the end.